fbpic/particles/injection/continuous_injection.py

```diff
@@ -181,9 +182,9 @@ def dens_func(x, y, z):
 
         # Create new particle cells
         # Determine the positions between which new particles will be created
-        Npz = self.nz_inject
         zmax = self.z_end_plasma
         zmin = self.z_end_plasma - self.nz_inject*self.dz_particles
+        Npz = 0 if (self.p_extent and (zmax < self.p_extent[0] or zmin > self.p_extent[1])) else self.nz_inject
```

Comment:
Could you first add an `if` condition here, to check that `p_extent` is not `None`?

Reply:
Hi Remi, sure I can add that. I thought the first test failing (when `self.p_extent` is `None`) would prevent the subsequent tests being performed, which I did try locally without any issues but perhaps this depends on the `python` version.